internal/tiltfile/tiltfile_state.go

```diff
@@ -623,6 +623,13 @@ func (s *tiltfileState) assembleK8sV2() error {
 	}
 
 	for workload, opts := range s.k8sResourceOptions {
+		if opts.nonWorkload {
+			r, err := s.makeK8sResource(opts.newName)
+			if err != nil {
+				return err
+			}
+			s.k8sByName[opts.newName] = r
```

Comment:
hmm what happens if you have an actual workload "foo" and a non-workload resource that you name "foo" -- will the latter overwrite the former in the `k8sByName` map?

Reply:
oh nvm this is handled at ln 403. (This is why i wasn't jazzed about splitting up the re/naming logic 😅 ) still, there should definitely be a test for this case, and it still feels kind of brittle to leave that map overwrite technically possible in this code.